engine/runtime-parser/src/main/java/org/enso/compiler/core/IR.java

```diff
@@ -103,6 +104,27 @@ default String pretty() {
   /** Storage for compiler diagnostics related to the IR node. */
   DiagnosticStorage diagnostics();
 
+  /**
+   * Get all diagnostic info associated with this IR node.
+   *
+   * @return the list of diagnostics
+   */
+  List<Diagnostic> diagnosticsList();
+
+  /**
+   * Get the copy of diagnostic storage associated with this IR node.
+   *
+   * @return the copy of diagnostic storage
+   */
+  DiagnosticStorage diagnosticsCopy();
+
+  /**
+   * Get storage for compiler diagnostics associated with this IR node.
+   *
+   * @return the diagnostic storage of this node
+   */
+  DiagnosticStorage getDiagnostics();
```

Comment:
There already is `diagnostics()`. Why do we need another `getDiagnostics()`?

Reply:
One can return null. I updated the docs